server/app/boards.go

```diff
@@ -338,46 +336,83 @@ func (a *App) PatchBoard(patch *model.BoardPatch, boardID, userID string) (*mode
 		if *patch.ChannelID != "" {
-			// TODO: this needs translated when available on the server
-			message := fmt.Sprintf(linkBoardMessage, username, updatedBoard.Title, boardLink)
-			err := a.store.PostMessage(message, "", *patch.ChannelID)
-			if err != nil {
-				a.logger.Error("Unable to post the link message to channel", mlog.Err(err))
-			}
+			a.postChannelMessage(fmt.Sprintf(linkBoardMessage, username, updatedBoard.Title, boardLink), updatedBoard.ChannelID)
 		} else if *patch.ChannelID == "" {
-			message := fmt.Sprintf(unlinkBoardMessage, username, updatedBoard.Title, boardLink)
-			err := a.store.PostMessage(message, "", oldChannelID)
-			if err != nil {
-				a.logger.Error("Unable to post the link message to channel", mlog.Err(err))
-			}
+			a.postChannelMessage(fmt.Sprintf(unlinkBoardMessage, username, updatedBoard.Title, boardLink), oldChannelID)
 		}
 	}
 
+	// Broadcast Messages to affected users
 	a.blockChangeNotifier.Enqueue(func() error {
 		a.wsAdapter.BroadcastBoardChange(updatedBoard.TeamID, updatedBoard)
-		if patch.ChannelID != nil && *patch.ChannelID != "" {
+
+		if patch.ChannelID != nil {
+			if *patch.ChannelID != "" {
+				members, err := a.GetMembersForBoard(updatedBoard.ID)
+				if err != nil {
+					a.logger.Error("Unable to get the board members", mlog.Err(err))
+				}
+				for _, member := range members {
+					if member.Synthetic {
+						a.wsAdapter.BroadcastMemberChange(updatedBoard.TeamID, member.BoardID, member)
+					}
+				}
+			} else if *patch.ChannelID == "" {
+				members, err := a.GetMembersForBoard(oldChannelID)
+				if err != nil {
+					a.logger.Error("Unable to get the board members", mlog.Err(err))
+				}
+				for _, oldMember := range members {
+					if oldMember.Synthetic {
+						a.wsAdapter.BroadcastMemberDelete(updatedBoard.TeamID, boardID, oldMember.UserID)
+					}
+				}
+			}
+		}
+
+		if patch.Type != nil && isTemplate {
 			members, err := a.GetMembersForBoard(updatedBoard.ID)
 			if err != nil {
 				a.logger.Error("Unable to get the board members", mlog.Err(err))
 			}
-			for _, member := range members {
-				if member.Synthetic {
-					a.wsAdapter.BroadcastMemberChange(updatedBoard.TeamID, member.BoardID, member)
-				}
-			}
-		} else if patch.ChannelID != nil && *patch.ChannelID == "" {
-			for _, oldMember := range oldMembers {
-				if oldMember.Synthetic {
-					a.wsAdapter.BroadcastMemberDelete(updatedBoard.TeamID, boardID, oldMember.UserID)
-				}
-			}
+			a.broadcastTeamUsers(updatedBoard.TeamID, updatedBoard.ID, *patch.Type, members)
 		}
 		return nil
 	})
 
 	return updatedBoard, nil
 }
 
+func (a *App) postChannelMessage(message, channelID string) {
+	err := a.store.PostMessage(message, "", channelID)
+	if err != nil {
+		a.logger.Error("Unable to post the link message to channel", mlog.Err(err))
+	}
+}
+
+func (a *App) broadcastTeamUsers(teamID, boardID string, boardType model.BoardType, members []*model.BoardMember) {
```

Comment:
It took me a bit to understand the usage of this method, so I'd suggest to add a docline stating the preconditions, something like:

```suggestion
// broadcastTeamUsers notifies the members of a team when a template changes its type
// from public to private or viceversa
func (a *App) broadcastTeamUsers(teamID, boardID string, boardType model.BoardType, members []*model.BoardMember) {
```